src/Elm/Ast.hs

```diff
@@ -85,6 +85,7 @@ data ElmPrim
     | ElmPair !TypeRef !TypeRef             -- ^ @(A, B)@ in elm
     | ElmTriple !TypeRef !TypeRef !TypeRef  -- ^ @(A, B, C)@ in elm
     | ElmList !TypeRef                      -- ^ @List A@ in elm
+    | ElmNonEmptyPair !TypeRef              -- ^ @NonEmpty A@ represented by @(A, List A)@ in elm
```

Comment:
We need this extra constructor in AST if we want to represent `NonEmpty` as JSON array. Reusing `ElmPair` would force use to represent `1 :| [ 2, 3]` as `[1, [2,3] ]` in JSON which would be a breaking change for existing APIs.

Reply:
How is `ElmPair` represented in JSON?

Reply:
as an array so `[1]` for singleton or `[1,2,3]` - so flat array.